### PR TITLE
logictest: replace mixed-version config versions

### LINKED ISSUE #90178: logictest: mixed-version config versions need to be bumped on master
Presently, we have `local-mixed-22.1-22.2` on master, yet with the 22.2 release, we should instead have `local-mixed-22.2-23.1`.

Jira issue: CRDB-20623

### PR DESCRIPTION
**Pull request by postamar**
The local-mixed-22.1-22.2 config is replaced with local-mixed-22.2-23.1.

Fixes #90178.

Release note:  None

### CODE REVIEW

**Comment by ajwerner** [comment]
@fqazi do you know how to test this? On the surface it looks right to me but I don't have in my head all the places that have to line up for this to do what we want.

**Comment by fqazi** [comment]
> @fqazi do you know how to test this? On the surface it looks right to me but I don't have in my head all the places that have to line up for this to do what we want.

We can kick off the nightly pipeline on this PR for the corpus generation. I kicked one off here: https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_ScratchProjectPutTcExperimentsInHere_SqlLogicDeclarativeSchemaChangerCorpusBazel/7028952?showRootCauses=false

**Review by fqazi — APPROVED** [review]
:LGTM:, but let's see if the manual nightly run is happy, which will at least ensure that a corpus can be generated. Afterwards, we can manually feed the corpus into a downlevel build or let the roachtest detect issues automatically once this is merged.

> Reviewed 16 of 16 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/90177#-:-NElUpTT8OnhnMRk8UNw:bvwygii)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @ajwerner)

**Comment by postamar** [comment]
Thanks for taking a look!

bors r+

**Comment by postamar** [comment]
bors cancel

File: build/teamcity/cockroach/nightlies/sqllogic_corpus_nightly_impl.sh
Comment (fqazi) [inline build/teamcity/cockroach/nightlies/sqllogic_corpus_nightly_impl.sh:78]: Ugh, this is an existing issue 🤦 , but there is a missing done on this loop. The corpus is generated but it messes up the build process status

**Comment by postamar** [comment]
bors cancel

File: build/teamcity/cockroach/nightlies/sqllogic_corpus_nightly_impl.sh
Comment (postamar) [reply build/teamcity/cockroach/nightlies/sqllogic_corpus_nightly_impl.sh:78]: fixed

**Pull request merged**
